flow/StreamCipher.h

```diff
@@ -73,7 +87,7 @@ class EncryptionStreamCipher final : NonCopyable, public ReferenceCounted<Encryp
 	StreamCipher cipher;
 
 public:
-	EncryptionStreamCipher(const StreamCipher::Key& key, const StreamCipher::IV& iv);
+	EncryptionStreamCipher(const StreamCipherKey* const key, const StreamCipher::IV& iv);
```

Comment:
```suggestion
	EncryptionStreamCipher(const StreamCipherKey* key, const StreamCipher::IV& iv);
```

Reply:
Idea here is to make "Key" "Const pointer pointing to a const value", any reason we want it to be "pointer to const value" ?

Reply:
Passing a const pointer puts unnecessary restrictions on the implementation IMO, since the scope of the pointer is only within the implementation.

Reply:
I feel making the API contract strict may be OK in this case, especially given we are dealing with EncryptionKeys like sensitive data; unless you feel strong about it, would like to keep it as is. 
Please let me know.

Reply:
The API contract should be strict, but that is achieved by using a pointer to a constant key. Making the pointer itself constant only puts restrictions on the implementation, it doesn't affect the API at all.